### ISSUE TITLE
Feature request: wait for asynchronous actions

### ISSUE THREAD

**Issue by martikaljuve** [amzn/style-dictionary#725]
Related issue: https://github.com/amzn/style-dictionary/issues/51

Would it be feasible to consider refactoring [performActions](https://github.com/amzn/style-dictionary/blob/main/lib/performActions.js) to wait for async actions?

My use case: I am using StyleDictionary actions to run linters on generated files. Some of these linters (e.g. stylelint, eslint) only have asynchronous functions available. I would now like to do some more processing on the linted files, but it looks like the next action will run before the linters have finished writing to disk.

Maybe there is a better place to add async linting in StyleDictionary that I haven't considered. Some workarounds that I could think of: 
- Combining linting action + next action into a single action
- Running linting + next action outside of StyleDictionary (e.g. `npm run build-dictionary && npm run lint && npm run ...`)

**Comment by dbanksdesign** [comment]
Thanks for bringing this issue up! I could see supporting a configuration like `synchronousActions: true | false` where the user could choose if they want actions to be performed synchronously or asynchronously. I do think the use case of running a linter on the generated files sounds better to be performed outside style dictionary though.

**Comment by martikaljuve** [comment]
Yeah, I guess running the linting outside of StyleDictionary is the easiest way for me now. 👍 

In terms of backwards-compatibility, a flag like `synchronousActions` might not even be necessary - e.g. if `performActions` used a chain of Promise.then calls instead, then existing synchronous actions (and async actions that _don't return a Promise_) would work the same, but async actions (ones that do return a Promise) would now be waited on.

```js
function performActions(dictionary, platform) {
  if (platform.actions) {
    let chain = Promise.resolve();
    for (const action of platform.actions) {
      chain = chain.then(() => action.do(dictionary, platform));
    }
  }
}
```

Same with `async`/`await`:

```js
async function performActions(dictionary, platform) {
  if (platform.actions) {
    for (const action of platform.actions) {
      await action.do(dictionary, platform);
    }
  }
}
```

**Comment by martikaljuve** [comment]
By the way, I omitted the other custom action that I would like to run from the first post, but I'll elaborate on it now.

We create a bunch of JS/JSON/TS/D.TS/CSS files for multiple themes, and most of them have a header generated with [formatHelpers.fileHeader](https://github.com/amzn/style-dictionary/blob/main/lib/common/formatHelpers/fileHeader.js). I would now like to "discard" the files where only the "Generated on <datetime>" in the header got changed, but not the actual contents. I was planning to create an Action that checks how many lines changed in the files (e.g. with `git diff`) and resets them, but I guess this can run as a separate script as well.

**Comment by thomasmattheussen** [comment]
I'm bumping into the same issue. Would be very nice if we get the option to `await` async actions.

**Comment by thomasmattheussen** [comment]
I'm bumping into the same issue. Would be very nice if we get the option to `await` async actions.
